### PR TITLE
User mentions

### PR DESCRIPTION
**Pull request by sanjanadabbiru**
Added controller and dao for user mentions

### CODE REVIEW

**Comment by aishakothare31** [comment]
looks good to me after the uid fix

File: controllers/UserMentionsController.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
+/**
+ * @file Controller RESTful Web service API for messages resource
```
Comment (shashwatrathod) [inline controllers/UserMentionsController.ts]: The docs should reflect that this file relates to UserMentions

File: controllers/UserMentionsController.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+ import { Request, Response, Express, json } from "express";
+ import { Session } from "..";
+ import IUserMentionsController from "../interfaces/IUserMentionsController";
+ import UserMentionsDao from "../daos/UserMentionsDao";
+ import TuitDao from "../daos/TuitDao";
+ import UserDao from "../daos/UserDao";
+import Tuit from "../models/tuits/Tuit";
+ 
+ /**
+  * @class MessageController Implements RESTful Web service API for messages resource.
```
Comment (shashwatrathod) [inline controllers/UserMentionsController.ts]: Review JSDocs

File: controllers/UserMentionsController.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+         UserMentionsController.userMentionsController.findUserLikesTuit
+       );
+       
+     }
+     return UserMentionsController.userMentionsController;
+   };
+
+   private constructor() {}
+
+findUserLikesTuit = async (req: Request, res: Response) => {
```
Comment (shashwatrathod) [inline controllers/UserMentionsController.ts]: Refactor method names

File: controllers/UserMentionsController.ts [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    const userId =
+      req.params.uid === "me" && (req.session as Session).profile
+        ? (req.session as Session).profile._id
+        : req.params.uid;
+    const userName = UserMentionsController.userDao.findUserById(userId);
+
+    const str = new String('@ellenripley');
```
Comment (shashwatrathod) [inline controllers/UserMentionsController.ts]: This implementation only works for ellenripley. Need to generalize it for any user

File: daos/UserMentionsDao.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    * @return MessageDao
+    */
+   public static getInstance = (): UserMentionsDao => {
+     if (UserMentionsDao.userMentionsDao === null) {
+       UserMentionsDao.userMentionsDao = new UserMentionsDao();
+     }
+     return UserMentionsDao.userMentionsDao;
+   };
+
+   findUserLikesTuit = async (uid: string[]): Promise<Tuit[]> =>
```
Comment (shashwatrathod) [inline daos/UserMentionsDao.ts:26]: change method name

File: controllers/UserMentionsController.ts (hunk 1 above)
Comment (sanjanadabbiru) [reply controllers/UserMentionsController.ts]: Sorry about that.. didn't refactor... was going to but fell sick

File: controllers/UserMentionsController.ts (hunk 4 above)
Comment (sanjanadabbiru) [reply controllers/UserMentionsController.ts]: Had an issue with getting userid from user.

File: daos/UserMentionsDao.ts (hunk 5 above)
Comment (sanjanadabbiru) [reply daos/UserMentionsDao.ts:26]: done

**Pull request merged**